src/scss/_table.scss

```diff
@@ -1,40 +1,101 @@
-.jazz-table__bordered {
-  border: 1px solid jazz-theme(table-border-color);
-  border-collapse: collapse;
-  line-height: 1.25;
+.jazz-table {
+  font-family: jazz-theme(table-font);
+  font-style: normal;
+  font-size: px-to-rem(16);
+  line-height: px-to-rem(24);
+
+  // border:collapse causes an issue with the weight of the first line to be doubled
+  // so using border-spacing instead
+  border-spacing: 0px;
 }
 
-.jazz-table__bordered th,
-.jazz-table__bordered td {
-  border: 1px solid jazz-theme(table-border-color);
-  padding: px-to-rem(12);
+.jazz-table th,
+.jazz-table td {
+  padding: px-to-rem(8) px-to-rem(16);
+  font-size: px-to-rem(16);
+  vertical-align: top;
   text-align: left;
-  font-size: px-to-rem(14);
 }
 
-.jazz-table__bordered th {
-  background-color: jazz-theme(table-th-bgcolor);
+.jazz-table th {
+  border-bottom: 3px solid jazz-theme(table-th-border-color);
+  font-weight: bold;
+  background: jazz-theme(table-th-bgcolor);
   color: jazz-theme(table-th-color);
 }
 
-.jazz-table__borderless {
-  border-collapse: collapse;
-  line-height: 1.25;
+.jazz-table th[scope=row] {
+  border-bottom: 1px solid jazz-theme(table-row-sep-color);
+  font-weight: normal;
+  background: jazz-theme(table-row-bgcolor);
+  color: jazz-theme(table-row-color);
 }
 
-.jazz-table__borderless th,
-.jazz-table__borderless td {
+.jazz-table td {
   border-bottom: 1px solid jazz-theme(table-row-sep-color);
-  padding: px-to-rem(12);
-  text-align: left;
-  font-size: px-to-rem(14);
 }
 
-.jazz-table__borderless th {
-  border-bottom: 3px solid jazz-theme(accent);
+.jazz-table tr {
+  background: jazz-theme(table-row-bgcolor);
+  color: jazz-theme(table-row-color);
+  font-weight: normal;
+}
+
+.jazz-table tr:hover {
+  background: jazz-theme(table-row-hover-bgcolor);
+  color: jazz-theme(table-row-hover-color);
+}
+
+.jazz-table__bordered td {
```

Comment:
We have been using the BEM (Block, Element, Modifier) naming convention for classes (http://getbem.com/naming/).  Following this convention, the naming should be "jazz-table--bordered".